### PR TITLE
Allow for udev symlink following

### PR DESCRIPTION
**Pull request by Maelstrom96**
This PR should properly allow for SM attach to use self.path when VDI.attach is called, but in case the drive /dev/ device name changes, on SM start, it should load the vdi and get the new VDI path linked by the symlink.

Edit : I haven't done extensive testing with unplugging drives and swapping SATA ports, but a SR scan is working as expected, and attaching drives is working correctly with symlinks to /dev/disk/by-id/ devices.

Commands used to test:
```
mkdir /srv/pass_drives/
cd /srv/pass_drives/
ln -s /dev/disk/by-id/ata-WDC_WD120EFBX-68B0EN0_5QKP16YB
ln -s /dev/disk/by-id/ata-WDC_WD120EFBX-68B0EN0_5QKT00MB
ln -s /dev/disk/by-id/ata-WDC_WD120EFBX-68B0EN0_5QKT9B9B
xe sr-create name-label=Pass_Drives type=udev content-type=disk device-config:location=/srv/pass_drives
```

This patch only updates self.path if it's a symlink (which isn't a current use case) means that it really shouldn't change the behavior of existing usage in SM, and be low risk.

### CODE REVIEW

File: drivers/udevSR.py [hunk 1]
```diff
@@ -148,6 +148,9 @@ def load(self, location):
             iso8601 = time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime(s[stat.ST_CTIME]))
             self.sm_config['hotplugged_at'] = iso8601
 
+            if os.path.islink(self.path):
```
Comment (MarkSymsCtx) [inline drivers/udevSR.py:151]: I would say that the `if os.path.islink(self.path):` is unnecessary here and it should just always do `self.path = os.pth.realpath(self.path)`

**Review by MarkSymsCtx — CHANGES_REQUESTED** [review]

File: drivers/udevSR.py (hunk 1 above)
Comment (Maelstrom96) [reply drivers/udevSR.py]: Agreed and fixed - I guess I wanted to make sure the behavior of `os.path.realpath` wouldn't negatively affect the function for non-symlink situations, but it should just return the same thing if it's a true path.

**Comment by MarkSymsCtx** [comment]
It will need a rebase for the Github action CI to run successfully but otherwise this looks good.

**Review by MarkSymsCtx — DISMISSED** [review]

**Review by MarkSymsCtx — CHANGES_REQUESTED** [review]
I think your rebase went very wrong there.

**Comment by Maelstrom96** [comment]
Should be good to go now, sorry about that.

**Review by MarkSymsCtx — APPROVED** [review]

**Review by TimSmithCtx — APPROVED** [review]

**Pull request merged**